daemon/ZilliqaDaemon.cpp

```diff
@@ -15,95 +15,157 @@
+const string SYNCTYPE_OPT = "--synctype";
+const char* synctype_descr =
+    "0(default) for no, 1 for new, 2 for normal, 3 for ds, 4 for lookup, 5 "
+    "for node recovery, 6 for new lookup , 7 for ds guard node sync and 8 "
+    "for offline validation of DB";
+
+const string default_logPath = "/run/zilliqa/";
+const string SUSPEND_LAUNCH = "SUSPEND_LAUNCH";
+const string launch_zilliqa_script = "launch_zilliqa.py";
+const string upload_incr_DB_script = "upload_incr_DB.py";
+const string download_incr_DB_script = "download_incr_DB.py";
+const string auto_backup_script = "auto_backup.py";
+const string daemon_log = "daemon-log.txt";
 
 enum SyncType : unsigned int {
   NO_SYNC = 0,
   NEW_SYNC,
   NORMAL_SYNC,
   DS_SYNC,
   LOOKUP_SYNC,
+  RECOVERY_ALL_SYNC,
+  NEW_LOOKUP_SYNC,
+  GUARD_DS_SYNC,
+  DB_VERIF
 };
 
-const string logName = "epochinfo-00001-log.txt";
-
-string ReadLastLine(const string& filePath, [[gnu::unused]] ofstream& log) {
-  ifstream logFile;
+ZilliqaDaemon::ZilliqaDaemon(int argc, const char* argv[], std::ofstream& log)
+    : m_log(log),
+      m_logPath(default_logPath),
+      m_curPath(boost::filesystem::current_path().string()),
+      m_port(-1),
+      m_recovery(0),
+      m_nodeIndex(0),
+      m_syncType(0),
+      m_cseed(false) {
+  if (ReadInputs(argc, argv) != SUCCESS) {
+    m_log << "Failed to read inputs" << endl;
+    exit(EXIT_FAILURE);
+  }
 
-  logFile.open(filePath + "/" + logName);
+  m_log << argv[0];
 
-  if (logFile.is_open()) {
-    logFile.seekg(-1, ios_base::end);
-  } else {
-    return "";
+  for (int i = 1; i < argc; ++i) {
+    m_log << " " << argv[i];
   }
 
-  bool keepLooping = true;
+  m_log << endl;
+  KillProcess();
+  StartNewProcess();
+  StartScripts();
+}
 
-  while (keepLooping) {
-    char ch;
-    logFile.get(ch);
+void ZilliqaDaemon::MonitorProcess() {
+  const string name = programName[0];
 
-    if ((int)logFile.tellg() <= 1) {
-      logFile.seekg(0);
-      keepLooping = false;
-    } else if (ch == ']') {
-      keepLooping = false;
-    } else {
-      logFile.seekg(-2, ios_base::cur);
+  if (m_pids[name].empty()) {
+    m_log << ZilliqaDaemon::CurrentTimeStamp().c_str() << "Looking for new "
+          << name << " process..." << endl;
+    vector<pid_t> tmp = ZilliqaDaemon::GetProcIdByName(name);
+
+    for (const pid_t& i : tmp) {
+      m_died[i] = false;
+      m_pids[name].push_back(i);
+      m_log << ZilliqaDaemon::CurrentTimeStamp().c_str()
+            << "Started monitoring new process " << name << " with PiD: " << i
+            << endl;
     }
+
+    return;
   }
 
-  string lastLine = "";
-  getline(logFile, lastLine);
+  for (const pid_t& pid : m_pids[name]) {
+    // If sig is 0 (the null signal), error checking is performed but no signal
+    // is actually sent
+    int w = kill(pid, 0);
 
-  // log<<"Line :"<<lastLine<<endl;
+    if (w < 0) {
+      if (errno == EPERM) {
+        m_log << ZilliqaDaemon::CurrentTimeStamp().c_str()
```

Comment:
If the timestamp is needed for every logging call, why not creating a helper function to reduce the redundancy